stats/stats/src/lib.rs

```diff
@@ -1,14 +1,216 @@
+use entity::chart_data_int;
+use sea_orm::{
+    DatabaseConnection, DbBackend, DbErr, EntityTrait, FromQueryResult, JsonValue, Statement,
+};
+use stats_proto::blockscout::stats::v1::{ChartInt, Counters, PointInt};
+use std::collections::HashMap;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ReadError {
+    #[error("database error {0}")]
+    DB(#[from] DbErr),
+    #[error("chart {0} not found")]
+    NotFound(String),
+}
+
+fn parse_json(row: JsonValue) -> Option<(String, (String, u64))> {
+    let name = row.get("name")?.as_str()?;
+    let date = row.get("date")?.as_str()?;
+    let value = row.get("value")?.as_u64()?;
+    Some((name.into(), (date.into(), value)))
+}
+
+fn get_counter_from_data(
+    data: &HashMap<String, (String, u64)>,
+    name: &str,
+) -> Result<u64, ReadError> {
+    data.get(name)
+        .map(|(_, value)| *value)
+        .ok_or_else(|| ReadError::NotFound(name.into()))
+}
+
+pub async fn get_counters(db: &DatabaseConnection) -> Result<Counters, ReadError> {
+    let data = JsonValue::find_by_statement(Statement::from_string(
+        DbBackend::Postgres,
+        r#"
+        SELECT distinct on (charts.id) charts.name, data.date, data.value 
+            FROM "chart_data_int" "data"
+            INNER JOIN "charts"
+                ON data.chart_id = charts.id
+            WHERE charts.type = 'COUNTER'
+            ORDER BY charts.id, data.id DESC;
+        "#
+        .into(),
+    ))
+    .all(db)
+    .await?;
+    let data: HashMap<_, _> = data.into_iter().filter_map(parse_json).collect();
+    let counters = Counters {
+        total_blocks_all_time: get_counter_from_data(&data, "total_blocks_all_time")?,
+    };
+    Ok(counters)
+}
+
+#[derive(FromQueryResult)]
+struct DateValue {
+    date: NaiveDateTime,
+    value: i64,
+}
+
+pub async fn get_chart_int(db: &DatabaseConnection, name: &str) -> Result<ChartInt, DbErr> {
+    let data: Vec<DateValue> = chart_data_int::Entity::find()
+        .from_raw_sql(Statement::from_sql_and_values(
+            DbBackend::Postgres,
+            r#"
+            SELECT data.date, data.value 
+                FROM "chart_data_int" "data"
+                INNER JOIN "charts"
+                    ON data.chart_id = charts.id
+                WHERE charts.name = $1
+                ORDER BY data.date;
+            "#,
+            vec![name.into()],
+        ))
+        .into_model()
+        .all(db)
+        .await?;
+    let chart = data
+        .into_iter()
+        .map(|row| PointInt {
+            date: row.date.format("%d-%m-%Y").to_string(),
+            value: row.value as u64,
+        })
+        .collect();
+    Ok(ChartInt { chart })
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
+    use migration::MigratorTrait;
+    use pretty_assertions::assert_eq;
+    use sea_orm::{ConnectionTrait, Database};
+    use url::Url;
+
+    async fn init_db(name: &str) -> DatabaseConnection {
+        let db_url = std::env::var_os("DATABASE_URL")
```

Comment:
change to std::env::var
https://github.com/blockscout/blockscout-rs/pull/235#discussion_r1039751140